src/graph/graph.c

```diff
@@ -1078,32 +1053,27 @@ static void _BulkDeleteImplicitEdges(Graph *g, GrB_Matrix Mask) {
 	// update the transposed adjacency matrix
 	GrB_Matrix_apply(tadj, Mask, GrB_NULL, GrB_IDENTITY_BOOL, tadj, desc);
 
-	/* if we have individual transposed matrices, repeat all the above steps
-	 * with the transposed Mask */
-	bool maintain_transpose;
-	Config_Option_get(Config_MAINTAIN_TRANSPOSE, &maintain_transpose);
-	if(maintain_transpose) {
-		for(int i = 0; i < relation_count; i++) {
-			GrB_Matrix TR = Graph_GetTransposedRelationMatrix(g, i);
+	/* repeat all the above steps with the transposed Mask */
+	for(int i = 0; i < relation_count; i++) {
+		GrB_Matrix TR = Graph_GetTransposedRelationMatrix(g, i);
 
-			// reset mask descriptor
-			GrB_Descriptor_set(desc, GrB_MASK, GxB_DEFAULT);
+		// reset mask descriptor
+		GrB_Descriptor_set(desc, GrB_MASK, GxB_DEFAULT);
 
-			/* isolate implicit edges
-			 * A will contain all implicitly deleted edges from TR */
-			GrB_Matrix_apply(A, Mask, GrB_NULL, GrB_IDENTITY_UINT64, TR, desc);
+		/* isolate implicit edges
+			* A will contain all implicitly deleted edges from TR */
+		GrB_Matrix_apply(A, Mask, GrB_NULL, GrB_IDENTITY_UINT64, TR, desc);
 
-			// free each multi edge array entry in A
-			GxB_Matrix_apply_BinaryOp1st(A, GrB_NULL, GrB_NULL,
-										 _binary_op_delete_edges, thunk, A, GrB_NULL);
+		// free each multi edge array entry in A
+		GxB_Matrix_apply_BinaryOp1st(A, GrB_NULL, GrB_NULL,
+										_binary_op_delete_edges, thunk, A, GrB_NULL);
 
-			// clear the relation matrix
-			GrB_Descriptor_set(desc, GrB_MASK, GrB_COMP);
-			GrB_Descriptor_set(desc, GrB_MASK, GrB_STRUCTURE);
+		// clear the relation matrix
+		GrB_Descriptor_set(desc, GrB_MASK, GrB_COMP);
+		GrB_Descriptor_set(desc, GrB_MASK, GrB_STRUCTURE);
 
-			// remove every entry of TR marked by Mask
-			GrB_Matrix_apply(TR, Mask, GrB_NULL, GrB_IDENTITY_UINT64, TR, desc);
-		}
+		// remove every entry of TR marked by Mask
+		GrB_Matrix_apply(TR, Mask, GrB_NULL, GrB_IDENTITY_UINT64, TR, desc);
 	}
```

Comment:
combine this loop with the one above it,
use the descriptor to transpose the mask, let's talk about this online.